stream/src/public_key/mod.rs

```diff
@@ -48,35 +48,22 @@
-//! During the handshake (described above), a shared x25519 secret is established using a Diffie-Hellman Key Exchange. This
-//! x25519 secret is then used to create a ChaCha20-Poly1305 cipher for encrypting all messages exchanged with the peer.
-//!
-//! ChaCha20-Poly1305 nonces (12 bytes) are constructed such that the first bit indicates whether the sender is a dialer (1) or
-//! dialee (0). The rest of the first byte (next 7 bits) is unused (set to 0). The next 2 bytes are a `u16` iterator.
-//! The next 8 bytes are a `u64` sequence number. When the sequence reaches `u64::MAX`, the iterator is incremented and the
-//! sequence is reset to 0. This technique provides each sender with a channel duration of `2^80` frames
-//! (and automatically terminates when this number of frames has been sent). In the blockchain context, validators often maintain
-//! long-lived connections with each other and avoiding connection re-establishment (to reset iterator/sequence with a new cipher)
-//! is desirable. The final byte is unused (set to 0).
-//!
-//! ```text
-//! +---+---+---+---+---+---+---+---+---+---+---+---+
-//! | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 |10 |11 |
-//! +---+---+---+---+---+---+---+---+---+---+---+---+
-//! | D |It(u16)|         Sequence(u64)         | U |
-//! +---+---+---+---+---+---+---+---+---+---+---+---+
-//!
-//! D = Dialer/Dialee, U = Unused, It = Iterator
-//! ```
-//!
-//! _We use a combination of `u64` (sequence) and `u16` (iterator) instead of implementing `u80/u88` because
-//! CPUs provide native support for `u64` operations (which will always be faster than an implementation of a
-//! "wrapping add" over arbitrary bytes). With this technique, almost all operations (other than iterator
-//! increments every `2^64` frames) are just a basic `u64` increment._
-//!
-//! This simple coordination prevents nonce reuse (which would allow for messages to be decrypted) and saves a small amount of
-//! bandwidth (no need to send the nonce alongside the encrypted message). This "pedantic" construction of the nonce
-//! also avoids accidentally reusing a nonce over long-lived connections when setting it to be a small hash (as in XChaCha-Poly1305).
+//! During the handshake (described above), a shared x25519 secret is established using a
+//! Diffie-Hellman Key Exchange. This x25519 secret is then used to create a ChaCha20-Poly1305
+//! cipher for encrypting all messages exchanged with the peer.
+//!
+//! Each peer maintains a pair of ChaCha20-Poly1305 nonces (12 bytes), one for itself and one for
+//! the other. Each nonce is constructed using a counter that starts at either 1 for the dialer, or
+//! 0 for the dialee. For each message sent, the relevant counter is incremented by 2, ensuring that
+//! the two counters have disjoint nonce spaces. The nonce is the least-significant 12 bytes of the
+//! counter, encoded big-endian.
+//!
+//! This provides 2^95 unique nonces per sender, sufficient for over 1 trillion years at 1 billion
```

Comment:
nit: Can we add a note around the rationale for this (maintaining long-lived connections in p2p networks can help with stability)? When you are connected to a good peer, you want to stay connected to them as long as possible.